### ISSUE TITLE
Jsdoc typedef import causing spurious errors regarding duplicate definitions

### ISSUE THREAD

**Issue by arciisine** [microsoft/TypeScript#36045]
**TypeScript Version:**  3.7.4

**Search Terms:**
typedef, import, getter

**Code**
All of these files within the same folder.  Attached is a zip file with all the files listed below.
[issue.zip](https://github.com/microsoft/TypeScript/files/4028436/issue.zip)

*index.d.ts*
```typescript
declare class Sizable {
  get size(): number;
}

declare global {
  export interface Array<T> extends Sizable {}
}

export { };
```

*first.js*
```javscript
/** @typedef {import('./index')}  */
[].size;
```

*second.js*
```javscript
/** @typedef {import('./index')}  */
[].size;
```

*tsconfig.json*
```json
{
  "compilerOptions": {
    "allowJs": true,
    "checkJs": true,
    "outFile": "both.js"
  },
  "files": [
    "first.js",
    "second.js"
  ]
}
```

**Expected behavior:**
When compiling above, with `tsc` I expect the code to compile correctly.  The import should only be processed once as it would be with an `import` statement.

**Actual behavior:**
```bash
$ tsc
first.js:3:4 - error TS2300: Duplicate identifier 'size'.

3 [].size;
     ~~~~

  second.js:3:4
    3 [].size;
         ~~~~
    'size' was also declared here.

second.js:3:4 - error TS2300: Duplicate identifier 'size'.

3 [].size;
     ~~~~

  first.js:3:4
    3 [].size;
         ~~~~
    'size' was also declared here.


Found 2 errors.
```

**Additional Notes**: 
If I convert `get size` to `getSize`, the code compiles correctly.  It appears the primary issues is surrounding the accessor.

**Comment by weswigham** [comment]
An `@typedef` needs a name, eg:
```
/** @typedef {import('./index')} Idx  */
[].size;
```
makes an alias `Idx` equal to the shape of the module at `./index`.

Since yours doesn't have a name, it falls back to using the name of the attached expression - `[].size`, so `size`. This means you end up creating two globals named `size`. Since they're globals, they conflict. If you're just trying to include the file for type-side-effects, write
```js
/// <reference path=./index"" />
```
instead of the typedef.

**Comment by arciisine** [comment]
I don't think the triple-slash ref works in `.js` files (but I'm happy to be wrong).  I'm using the `@typedef` import as I was not able to get the reference tag to work.   The imports I'm pulling in are modifying the global type space so there is nothing specific to import. 

Also, changing the accessor to a proper method seems to resolve the issue, seeming to indicate that accessors are being treated differently.

I will try to add a random id as the type alias in the `@typedef` and see if that resolves my immediate issue.

**Comment by arciisine** [comment]
@weswigham So, I was definitely off on the triple slash, and that is a better path vs the jsdoc import.  

Per the issue above, still not sure why accessors are treated differently than regular methods.

**Comment by DanielRosenwasser** [comment]
This makes me think we need a no-nonsense `checkJs` mode, or that in `checkJs` we should error on clearly-bad behavior in JSDoc like this. (@sandersn)
